### PR TITLE
feat : show an effect when the player eat an item

### PR DESCRIPTION
**Pull request by doyoon323**

### CODE REVIEW

File: src/screen/GameScreen.java [hunk 1]
```diff
@@ -771,10 +768,21 @@ && checkCollision(bullet, this.enemyShipSpecial)) {
...
+						Thread thread = new Thread();
+						thread.start();
+						try{
+							for(int i=0;i<3;i++){
+								Thread.sleep(300);
+							}
+						} catch(Exception e){
+							e.printStackTrace();
+						}
+						 */
```
Comment (NiceGuy1313) [inline src/screen/GameScreen.java:785]: [P3]

There is no fault. In particular, it is good to leave the old code while processing comments out in this part.

**Review by NiceGuy1313 — APPROVED** [review]

File: src/screen/GameScreen.java [hunk 2]
```diff
@@ -790,8 +798,15 @@ && checkCollision(bullet, this.enemyShipSpecial)) {
 				if(item.getSpriteType() == SpriteType.PerpleEnhanceStone){
 					this.enhanceManager.PlusNumEnhanceStoneAttack(1);
 				}
-				this.ship.checkGetItem(item);
+				/* version 2.0 : when item is destroyed, show effect */
```
Comment (alicek0) [inline src/screen/GameScreen.java:801]: [P3] Good job leaving comments to help other developers understand the code.

File: src/entity/Item.java [hunk 3]
```diff
@@ -39,6 +41,7 @@ public Item(final int positionX, final int positionY) {
         item_dy = Math.random() > 0.5 ? 1 : -1;
         this.setSprite();
         isdetroyed = false;
+        this.destructionCooldown = Core.getCooldown(200);
```
Comment (alicek0) [inline src/entity/Item.java:44]: [P3] Good utilization of exsiting Cooldown class

File: src/screen/GameScreen.java [hunk 4]
```diff
@@ -790,8 +798,15 @@ && checkCollision(bullet, this.enemyShipSpecial)) {
...
+				if (!item.isDestroyed()) {
+					this.ship.checkGetItem(item);
+					item.destructionCooldown.reset();
+					this.logger.info("Get Item ");
+				}
 			}
+			if (item.isDestroyed() && item.destructionCooldown.checkFinished())
```
Comment (alicek0) [inline src/screen/GameScreen.java:808]: [P3] Smart way of showing item get effect for a certain amount of time using Cooldown

**Review by alicek0 — APPROVED** [review]
It must have taken effort to understand existing code. I think the way of displaying the effect when getting an item utilizing Cooldown was smart. Good job! 👍

**Comment by ghost** [comment]
I closed PR because it was a merge request to main.

**Pull request merged**